### PR TITLE
fluent 7 + driver cleanup

### PR DESCRIPTION
**Pull request by tannernelson**
Adds Fluent 0.7 support which includes Preparations.

Additionally removes `Driver` from protocol names and makes `Log` an instance.

### CODE REVIEW

File: Sources/Vapor/Fluent/Prepare.swift [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+                do {
+                    try hasPrepared = database.hasPrepared(preparation)
+                } catch {
+                    self.error("Failed to start preparation")
+                    throw CommandError.custom("\(error)")
+                }
+
+                if !hasPrepared {
+                    print("Preparing '\(name)'")
```
Comment (LoganWright) [inline Sources/Vapor/Fluent/Prepare.swift]: should this be a log instead of a print?

File: Sources/Vapor/Hash/SHAHasher.swift [hunk 2]
```diff
@@ -23,26 +25,37 @@ public class SHA2Hasher: HashDriver {
     }
 
     /**
+        HMAC key.
+    */
+    public var key: String {
```
Comment (LoganWright) [inline Sources/Vapor/Hash/SHAHasher.swift]: Should key be a variable thing? It seems like it should be constant for the live of the hasher

**Comment by LoganWright** [comment]
</img>

**Comment by LoganWright** [comment]
This also seems to solve the issue w/ #322

File: Sources/Vapor/Fluent/Prepare.swift (hunk 1 above)
Comment (tannernelson) [inline Sources/Vapor/Fluent/Prepare.swift]: This is calling `self.print()` on command, not `Swift.print()`

**Pull request merged**